pkg/infoschema/builder.go

```diff
@@ -969,8 +969,11 @@ func NewBuilder(r autoid.Requirement, factory func() (pools.Resource, error), in
 	}
 }
 
-func tableBucketIdx(tableID int64) int {
-	return int(tableID % bucketCount)
+func tableBucketIdx(tableID int64) (idx int) {
+	if idx = int(tableID % bucketCount); idx < 0 {
+		idx = -idx
```

Comment:
If I just read this line with no more context ... I would be feel puzzled why tableID could be a negative number ...

Reply:
You can see the issue: https://github.com/pingcap/tidb/issues/52012

Reply:
And maybe it would be a better way if we can track where negative tableID comes from, and return error for the illegal input ...

Reply:
In fact, this PR sees a negative `tableID` as an "legal" value even if we will not not create an table with a negative id. And pass a negative table id to `TableByID` still returns error because table not exist...

Another option is regarding negative id as "illegal", so we need also check the table id both in building a infoschema and `TableByID`, and returns error if it is negative instead of panic